File: chapter_03/3_3.py
num_list = [23, 12, 44, 23, 12, 44, 55, 77, 65, 92]

lower_1 = 10000
lower_2 = 10000

for num in num_list:
    if num < lower_1:
        lower_1 = num
    if (num < lower_2) and (num > lower_1):
        lower_2 = num

print(f"lower_2 : {lower_2}") # lower_2は23

"""
計算量はO(N)
"""

File: chapter_03/c++/3_3.cpp
#include <string>
#include <vector>
#include <iostream>
#include <climits>

using namespace std;

int main(){
    vector<int> list_a = {21, 11, 35, 64, 23, 56, 124, 23, 12};

    int lower_1 = INT_MAX;
    int lower_2 = INT_MAX;

    for (int i=0; i<list_a.size(); i++){
        if (list_a[i] < lower_1) lower_1 = list_a[i];
        else if (list_a[i] < lower_2) lower_2 = list_a[i];
    }

    cout << "2nd lower num : " << lower_2 << endl;
    /*
    計算量は、O(N)
    */
}